### ISSUE TITLE
[Feature Request] Defining your own callback

### ISSUE THREAD

**Issue by Interspatial** [comet-ml/issue-tracking#187]
Hi, 
Could you add support for defining your own callback, or to disable cometml's callback altogether? I want to log some extra details by replacing the default callback for an experiment. (particularly in fastai)

Many thanks

**Comment by dsblank** [comment]
Hi @Interspatial , are you referring to the keras callback? That only works with keras.

If you want to disable auto-logging in fastai, you'll need to set the `COMET_DISABLE_AUTO_LOGGING=1` like this:

```
COMET_DISABLE_AUTO_LOGGING=1 python /myprogram,.py
```

Or did I misunderstand?

**Comment by Interspatial** [comment]
@dsblank I'm referring to the fast.ai callback. I'm getting some strangely large losses being measured once I unfreeze my model, when as measured by fastai are much smaller(order of 500x), but that's another issue.

**Comment by dsblank** [comment]
@Interspatial Can you send me a code snippet, or DM me on slack with the the URL to your comet.ml experiment (I'm Doug over at https://cometml.slack.com/ )

There isn't a fastai callback in comet_ml.

**Comment by Interspatial** [comment]
Could it be internally using the pytorch callback? I just looked at the fastai example in your repo.
`
model = create_cnn(data, models.resnet152, ps=0.1, lin_ftrs=[256, 128], metrics=[accuracy])
experiment = (...)
### after training unfrozen for a bit ###
model.unfreeze()
lrs = np.array([5e-6, 5e-5, 5e-4])

model = model.to_fp16()
with experiment.train():
    for i in range(3):
        model.fit_one_cycle(1, max_lr=lrs)`

**Comment by dsblank** [comment]
Fastai does indeed use pytorch, and we auto-log pytorch's loss metric during training. In addition, fastai auto-logs any associated metric during training at the end of an epoch, and the current epoch at the beginning of an epoch. 

"Callback" is a technical term involving a function that is defined and then called later. Keras uses callbacks, and we have a special Keras callback for additional logging support. 

You can disable all auto-logging by setting the environment variable `COMET_DISABLE_AUTO_LOGGING=1`, and then you can log anything you like at any time using `Experiment.log_metric()`, `Experiment.log_parameter()`, or `Experiment.log_other()`. You can also just ignore the auto-logged values, and log your own.

Hope that helps! For more information, see: 

* https://medium.com/comet-ml/comet-ml-cheat-sheet-supercharge-your-machine-learning-experiment-management-7786e08f8e9e
* https://www.comet.ml/docs/python-sdk/Experiment/#experimentlog_metric

**Comment by dsblank** [comment]
Regarding the graph, that is one of the options from the UI when analyzing experiments in a project. Simply select "Parallel Coordinates" as the graph type, and select the hyperparameters to see along the x-axis. For more information, please visit our slack channel.
